### PR TITLE
Msg handler basic implementation

### PR DESCRIPTION
**Pull request by Kolezhniuk**

### CODE REVIEW

File: src/iden3comm/handlers/auth.ts [hunk 1]
```diff
@@ -133,6 +134,19 @@ export interface IAuthHandler {
     response: AuthorizationResponseMessage;
   }>;
 }
+
+type AuthReqOptions = {
+  did: DID;
```
Comment (volodymyr-basiuk) [inline src/iden3comm/handlers/auth.ts:139]: Is it better to name `senderDid` ?

File: src/iden3comm/handlers/contract-request.ts [hunk 2]
```diff
@@ -44,6 +45,12 @@ export type ContractInvokeHandlerOptions = {
   challenge?: bigint;
 };
 
+export type ContractMessageHandlerOptions = {
+  did: DID;
```
Comment (volodymyr-basiuk) [inline src/iden3comm/handlers/contract-request.ts:49]: same here, `senderDid` ?

File: src/iden3comm/handlers/message-handler.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { BasicMessage, IPackageManager } from '../types';
+import { AuthMessageHandlerOptions } from './auth';
+import { RevocationStatusMessageHandlerOptions } from './revocation-status';
+import { ContractMessageHandlerOptions } from './contract-request';
+
+export interface IProtocolMessageHandler {
```
Comment (volodymyr-basiuk) [inline src/iden3comm/handlers/message-handler.ts:6]: let's add comments for documentation here for public/exported types

File: src/iden3comm/handlers/revocation-status.ts [hunk 4]
```diff
@@ -14,6 +15,14 @@ import { TreeState } from '../../circuits';
 import { byteEncoder } from '../../utils';
 import { proving } from '@iden3/js-jwz';
 import { IIdentityWallet } from '../../identity';
+import { AbstractMessageHandler, IProtocolMessageHandler } from './message-handler';
+
+export type RevocationStatusMessageHandlerOptions = {
```
Comment (volodymyr-basiuk) [inline src/iden3comm/handlers/revocation-status.ts:20]: add comments here please

**Review by volodymyr-basiuk — APPROVED** [review]

File: src/iden3comm/handlers/auth.ts [hunk 5]
```diff
@@ -185,38 +221,17 @@ export class AuthHandler implements IAuthHandler {
...
-    if (opts.mediaType === MediaType.SignedMessage && !opts.packerOptions) {
-      throw new Error(`jws packer options are required for ${MediaType.SignedMessage}`);
-    }
-
-    if (authRequest.to) {
-      // override sender did if it's explicitly specified in the auth request
-      did = DID.parse(authRequest.to);
-    }
+    // override sender did if it's explicitly specified in the auth request
+    const did = authRequest.to ? DID.parse(authRequest.to) : ctx.did;
```
Comment (ilya-korotya) [inline src/iden3comm/handlers/auth.ts:233]: In comment above: "override sender did ...", but here you are overriding target did.

**Review by ilya-korotya — COMMENTED** [review]
Could you recheck the documentation comments for the function processZeroKnowledgeProofRequests in handlers/common file, please? Because, realte to the documentation we should pass the sender of the message twice:
```
 * @param senderIdentifier - The identifier of the **sender**.
 * @param requests - An array of zero knowledge proof requests.
 * @param from - The identifier of the **sender**.
 ```

**Comment by ilya-korotya** [comment]
@Kolezhniuk did you implement handlers for proof request/response? There is a handler only for contract invoke.

https://iden3-communication.io/proofs/1.0/request/
https://iden3-communication.io/proofs/1.0/response/

**Review by volodymyr-basiuk — DISMISSED** [review]

**Review by vmidyllic — DISMISSED** [review]

**Comment by vmidyllic** [comment]
> > @Kolezhniuk did you implement handlers for proof request/response? There is a handler only for contract invoke.
> > https://iden3-communication.io/proofs/1.0/request/ https://iden3-communication.io/proofs/1.0/response/
> 
> @ilya-korotya feel free to contribute

no need in that. It is probably must be deprecated

**Review by volodymyr-basiuk — APPROVED** [review]

**Review by volodymyr-basiuk — APPROVED** [review]

**Review by vmidyllic — APPROVED** [review]

**Pull request merged**